src/librbd/io/ImageRequest.cc

```diff
@@ -511,7 +522,7 @@ template <typename I>
 void ImageDiscardRequest<I>::prune_object_extents(ObjectExtents &object_extents) {
   I &image_ctx = this->m_image_ctx;
   CephContext *cct = image_ctx.cct;
-  if (!cct->_conf->rbd_skip_partial_discard) {
+  if (!cct->_conf->rbd_skip_partial_discard || this->m_force_discard) {
```

Comment:
Sorry -- thinking about this a little more, I think that the ```ImageDiscardRequest``` should just take a "skip_partial_discard" boolean instead of "force" discard. 

There is already a tracker ticket for recording the state of "skip partial discard" in the journal event [1], so seeing your change to record the "force discard" in the journal event reminded me of the issue.

The "writesame" API methods would always pass "false" -- other callers to discard would pass the contents of "ImageCtx::skip_partial_discard" (which would need to be added like other config params are stored in ImageCtx).

[1] http://tracker.ceph.com/issues/16571